custom_components/hafas/utils.py

```diff
@@ -49,6 +50,18 @@ def to_dict(item):
                 "stopovers":        to_dict(item.stopovers or []),
             }
 
+        case StationBoardLeg():
```

Comment:
Just a remark: This doesn't seem to have a `mode`. I'd it is not critical, unless someone asks for it. To get it, apparently another call to the API would need to be made for each leg...